README.md
zabbix Cookbook
===============
This cookbook is to install zabbix-server 3.0 LTS and zabbix-agent LTS on RHEL 7. Using MariaDB (Mysql) database.

Requirements
------------
#### Platform
OS Platform: RHEL/CentOS 7

Attributes
----------
#### zabbix::server
<table>
  <tr>
    <th>Key</th>
    <th>Type</th>
    <th>Description</th>
    <th>Default</th>
  </tr>
  <tr>
    <td><tt>['zabbix']['server']['enabled']</tt></td>
    <td>Boolean</td>
    <td>Will install zabbix server if true</td>
    <td><tt>false</tt></td>
  </tr>
  <tr>
    <td><tt>['zabbix']['php']['server_name']</tt></td>
    <td>Text</td>
    <td>Zabbix server hostname for php</td>
    <td><tt>empty</tt></td>
  </tr>
  <tr>
    <td><tt>['zabbix']['server']['conf']['DBPassword']</tt></td>
    <td>Text</td>
    <td>DB password located in the zabbix-server.conf file</td>
    <td><tt>empty</tt></td>
  </tr>
  <tr>
    <td><tt>['zabbix']['php']['timezone']</tt></td>
    <td>Text</td>
    <td>Set your local timezone here</td>
    <td><tt>Europe/Riga</tt></td>
  </tr>
</table>

#### zabbix::mariadb
<table>
  <tr>
    <th>Key</th>
    <th>Type</th>
    <th>Description</th>
    <th>Default</th>
  </tr>
  <tr>
    <td><tt>['zabbix']['mysql']['root_password']</tt></td>
    <td>Text</td>
    <td>Root DB password</td>
    <td><tt>empty</tt></td>
  </tr>
</table>

#### zabbix::agent
<table>
  <tr>
    <th>Key</th>
    <th>Type</th>
    <th>Description</th>
    <th>Default</th>
  </tr>
  <tr>
    <td><tt>['zabbix']['agent']['Hostname']</tt></td>
    <td>Text</td>
    <td>Sets the Hostname setting in zabbix-agent.conf</td>
    <td><tt>empty</tt></td>
  </tr>
  <tr>
    <td><tt>['zabbix']['agent']['enabled']</tt></td>
    <td>Boolean</td>
    <td>Will install zabbix agent if true</td>
    <td><tt>false</tt></td>
  </tr>
  <tr>
    <td><tt>['zabbix']['agent']['Server']</tt></td>
    <td>Text</td>
    <td>Sets the Server setting in zabbix-agentd.conf</td>
    <td><tt>127.0.0.1</tt></td>
  </tr>
  <tr>
    <td><tt>['zabbix']['agent']['ServerActive']</tt></td>
    <td>Text</td>
    <td>Sets the ServerActive setting in zabbix-agentd.conf</td>
    <td><tt>127.0.0.1</tt></td>
  </tr>
</table>

Usage
-----
#### zabbix::default

Create a role or environment files containing similar info:

```ruby
name 'zabbix_server'

description 'This role is meant to be applied to the zabbix server'

run_list 'recipe[zabbix]'

override_attributes(
  zabbix: {
    php: {
      servername: 'zabbix.domain.tld'
    },
  agent: {
      version: '2.4.6',
      servers: 'zabbix.domain.tld',
      servers_active: 'zabbix.domain.tld',
      Hostname: 'zabbix.domain.tld',
      enabled: 'true'
    },
    mysql: {
      root_password: 'XXXXXXXXXX',
      password: 'XXXXXXXXXX'
    },
    server: {
      version: '3.0.0',
      enabled: 'true',
      conf: {
        DBPassword: 'XXXXXXXXX'
      }
    }
  }
)
```

Contributing
------------
Not open to contribution at this time.

License and Authors
-------------------
Authors: Som3guy

CHANGELOG.md
zabbix CHANGELOG
================

This file is used to list changes made in each version of the zabbix cookbook.

1.1.0
-----

- Som3guy - Installs zabbix 3.0.0

1.0.0
-----
- Som3guy - Working installs of zabbix 2.4.x

0.1.0
-----
- Som3guy - Initial release of zabbix

- - -
Check the [Markdown Syntax Guide](http://daringfireball.net/projects/markdown/syntax) for help with Markdown.

The [Github Flavored Markdown page](http://github.github.com/github-flavored-markdown/) describes the differences between markdown on github and standard markdown.


recipes/server.rb
# Install the zabbix 2.4 repo

   execute 'zabbix_repo_install' do
      command 'rpm -ivh http://repo.zabbix.com/zabbix/3.0/rhel/7/x86_64/zabbix-release-3.0-1.el7.noarch.rpm'
      not_if { File.exist?("#{node['zabbix']['server']['repo']}")}
   end

package ['zabbix-server-mysql', 'zabbix-web-mysql']  do
	version "#{node['zabbix']['server']['version']}-#{node['zabbix']['server']['release']}"
    action :install
end


mysqluser = node['zabbix']['mysql']['user']
mysqlpassword = node['zabbix']['mysql']['password']
unless Dir.exist? "#{node['zabbix']['mysql']['zabbix_dir']}"
   execute 'prep_mysql_db' do
	   command "mysql -e \"create database zabbix character set utf8 collate utf8_bin; grant all privileges on zabbix.* to #{mysqluser}@localhost identified by '#{mysqlpassword}';\""
   end
end

unless File.exist? "#{node['zabbix']['mysql']['zabbix_dir']}/maintenances.frm"
   execute 'mysql_import_schema' do
	   command "mysql -u#{node['zabbix']['mysql']['user']} -p#{node['zabbix']['mysql']['password']} zabbix < #{node['zabbix']['mysql']['zabbix_prep_dir']}#{node['zabbix']['server']['version']}/create/schema.sql"
   end

   execute 'mysql_import' do
	   command "mysql -u#{node['zabbix']['mysql']['user']} -p#{node['zabbix']['mysql']['password']} zabbix < #{node['zabbix']['mysql']['zabbix_prep_dir']}#{node['zabbix']['server']['version']}/create/images.sql"
   end

   execute 'mysql_import_data' do
	    command "mysql -u#{node['zabbix']['mysql']['user']} -p#{node['zabbix']['mysql']['password']} zabbix < #{node['zabbix']['mysql']['zabbix_prep_dir']}#{node['zabbix']['server']['version']}/create/data.sql"
   end
end

directory '/etc/zabbix/zabbix_server.conf.d' do
  owner 'root'
  group 'zabbix'
  mode '0755'
  action :create
end



template 'zabbix_server.conf' do
	path '/etc/zabbix/zabbix_server.conf'
	source 'zabbix_server.conf.erb'
	mode 0640
	owner 'root'
	group 'zabbix'
end

template 'zabbix.conf' do
	path '/etc/httpd/conf.d/zabbix.conf'
	source 'zabbix.conf.erb'
	mode 0644
	owner 'root'
	group 'root'
end

template 'zabbix.conf.php' do
	path '/etc/zabbix/web/zabbix.conf.php'
	source 'zabbix.conf.php.erb.html'
	mode 0644
	owner 'root'
	group 'root'
end

template 'selinux_config' do
	path '/etc/selinux/config'
	source 'zabbix_selinux.erb'
	mode 0644
	owner 'root'
	group 'root'
end

iptables_rule 'zabbix_firewall' do
	action :enable
end

service 'iptables' do
	action [:enable, :restart]
end

service 'zabbix-server' do
	action [:enable, :restart]
end

service 'httpd' do
	action [:enable, :restart]
end

